### PR TITLE
Check chain hash for channel announcement and update

### LINKED ISSUE #2228: Fail to apply gossip update for invalid chain hash
Currently, [`NetworkGraph::update_channel_unsigned`](https://docs.rs/lightning/latest/lightning/routing/gossip/struct.NetworkGraph.html#method.update_channel_unsigned) and similar methods will happily apply updates, even if they don't match its `genesis_hash`. This for example happens when syncing a testnet graph with the mainnet RGS url. 

However, we should check the given chain hash in `NetworkGraph::update_channel_intern` and refuse to apply any updates that don't match the network we're on.

### PR DESCRIPTION
**Pull request by freddiekrugerrand**
Fixes #2228

### CODE REVIEW

File: lightning/src/routing/gossip.rs [hunk 1]
```diff
@@ -2311,6 +2325,15 @@ pub(crate) mod tests {
 			Ok(_) => panic!(),
 			Err(e) => assert_eq!(e.err, "Channel announcement node had a channel with itself")
 		};
+
+		// Test that channel announcements with the wrong chain hash are ignored.
```
Comment (dunxen) [inline lightning/src/routing/gossip.rs:2329]: nit: Could we just include why it's the wrong chain hash here? I got a little confused with it in isolation but remembered we were dealing with testnet in these gossip tests.

"testnet vs mainnet" or something similar.

**Review by dunxen — COMMENTED** [review]
LGTM with one small nit.

**Comment by benthecarman** [comment]
This could also be done for rapid gossip sync updates

File: lightning/src/routing/gossip.rs [hunk 2]
```diff
@@ -1840,6 +1847,13 @@ impl<L: Deref> NetworkGraph<L> where L::Target: Logger {
 				return Err(LightningError{err: "Couldn't find channel for update".to_owned(), action: ErrorAction::IgnoreError});
 			},
 			Some(channel) => {
+				if msg.chain_hash != self.genesis_hash {
```
[the remark this answers is not in the document]
Comment (dunxen) [reply lightning/src/routing/gossip.rs:1850]: True, we can return right at the start.

**Comment by freddiekrugerrand** [comment]
> This could also be done for rapid gossip sync updates

Took a look at `processing.rs` to make sure RGS is fully covered:
- `update_channel_unsigned`: calls `update_channel_intern` which has check
- `add_channel_from_partial_announcement`: no `chain_hash` provided

Wondering whether we should add a top level check [here](https://github.com/lightningdevkit/rust-lightning/blob/main/lightning-rapid-gossip-sync/src/processing.rs#L70) in RGS so that we can exit early if we have the wrong hash? Would need to add `get_genesis_hash` (or `check_genesis_hash`) to `NetworkGraph`.

**Pull request merged**